### PR TITLE
release-notes.txt: add notes for 2019.07 release

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Release notes for 2019.07 release

### Testing procedure

Please read through, and check that, for the area of the OS which you're familiar with, that the existing information is correct, and there are no omissions.

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,550 @@
...
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2019.07 release includes a number of new features including many new
+boards and cpu, riotboot added to many new and old boards, usb is now available,
```
Comment (miri64) [inline release-notes.txt]:
```suggestion
boards and cpu, riotboot added to many new and old boards, USB is now available,
```

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,550 @@
...
+ble improvements, ethernet on stm32 platforms, progress in gnrf fragmentation,
```
Comment (miri64) [inline release-notes.txt]:
```suggestion
BLE improvements, Ethernet on stm32 platforms, progress in GNRC fragmentation,
```

However GNRC fragmentation is a) very unspecific b) not in yet.

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,550 @@
...
+as well as many bug fixed and documentation updates.  Testing has also improved
+with both On-Target testing increasing and now Hardware Assisted automated tests
```
Comment (miri64) [inline release-notes.txt]: either

```suggestion
with both On-Target Testing increasing and now Hardware Assisted Automated Tests
```

or

```suggestion
with both on-target testing increasing and now hardware assisted automated tests
```

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,550 @@
...
+
+
+Notations used below:
+=====================
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+New features and changes
+========================
```
Comment (miri64) [inline release-notes.txt]: Many things below are rather bug fixes...

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,550 @@
...
+    * tests/unittests: split remaining packages tests to different test
+                       directory
+
+API Changes
+========================
+- uart_stdio RX is now optional
+- CPU_MODEL declared in boards/Makefile.features
+
+
+```c
```
Comment (miri64) [inline release-notes.txt]: Do we have to be as detailed? and isn't this rather

~~~md
```diff
~~~

?

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,550 @@
...
+#10800: iotlab-m3: thread tests failing
+#10559: I2C API write_regs does not fit implementation
+#10468: Tinycryt upstream rewrote history in master branch
+#8653: msba2: default example fails on assert
+#6874: SAMD21: possible CMSIS bug ?
+#4612: pkg: TLSF does not build for 16 bit platforms
+
+Acknowledgements
+================
+We would like to thank all companies (vendors) that provided us with (their)
```
Comment (miri64) [inline release-notes.txt]: Template much? ;-).

File: release-notes.txt (hunk 5 above)
Comment (MrKevinWeiss) [reply release-notes.txt]: Well I wanted the c syntax highlighted but I can also go with diff if you prefer.  I think I would like to note this though.

File: release-notes.txt (hunk 6 above)
Comment (MrKevinWeiss) [reply release-notes.txt]: (vendors) is my favorite company!

File: release-notes.txt (hunk 4 above)
Comment (MrKevinWeiss) [reply release-notes.txt]: bug fix or change?

**Comment by MrKevinWeiss** [comment]
updated with fixes I agreed with...

File: release-notes.txt [hunk 7]
```diff
@@ -510,7 +480,7 @@ Fixed Issues from the last release (2019.04)
 
 Acknowledgements
 ================
-We would like to thank all companies (vendors) that provided us with (their)
+We would like to thank all companies that provided us with
```
Comment (miri64) [inline release-notes.txt]: Now the formatting of this paragraph is off

Comment (MrKevinWeiss) [reply release-notes.txt]: I didn't think it was that important.  All fixed.

File: release-notes.txt (hunk 4 above)
Comment (miri64) [reply release-notes.txt]: For bug fixes we have the "issues fixed" section. Non-critical issues that came up during the release cycle are usually not worth mentioning.

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,520 @@
...
+    + usbus: Add CDC-ECM (Ethernet Control Model) function
+    * usbus: simplify adding entry to list
+    * sys/stdio_ethos: replace USE_ETHOS_FOR_STDIO by stdio_ethos pseudomodule
+
+Networking
+----------
+    * gnrc_ipv6_nib: add address from netif to address validation timer
+    + netdev_ieee802154: add txpower and page
+    * gnrc_sixlowpan_frag: allow send of multiple datagrams simultaneously
+    + gnrc_sixlowpan_frag: Initial import of a fragment size hint feature
```
Comment (miri64) [inline release-notes.txt]: Since this is just prep for #11068, I don't think this is worth mentioning.

File: release-notes.txt (hunk 5 above)
Comment (miri64) [reply release-notes.txt]: kk

File: release-notes.txt [hunk 9]
```diff
@@ -1,3 +1,520 @@
...
+        and no error
+#11043: boards/z1: tests/rng resets board
+#10927: gnrc_tftp: Missing minimum length check
+#10878: nrfmin can get stuck and never reach RX (while TX works)
+#10800: iotlab-m3: thread tests failing
+#10559: I2C API write_regs does not fit implementation
+#10468: Tinycryt upstream rewrote history in master branch
+#8653: msba2: default example fails on assert
+#6874: SAMD21: possible CMSIS bug ?
+#4612: pkg: TLSF does not build for 16 bit platforms
```
Comment (miri64) [inline release-notes.txt]: Was this really fixed or just closed as "won't fix"?

**Comment by MrKevinWeiss** [comment]
Alrighty, did a bit of cleanup.  Maybe that is good but also I don't mind pushing the release until tomorrow.  A bit too hot to think clearly...

File: release-notes.txt [hunk 10]
```diff
@@ -1,3 +1,514 @@
...
+    + sys/bluetil: add bluetil_addr_from_str()
+    + usbus: Add CDC-ECM (Ethernet Control Model) function
+    * usbus: simplify adding entry to list
+    * sys/stdio_ethos: replace USE_ETHOS_FOR_STDIO by stdio_ethos pseudomodule
+
+Networking
+----------
+    * gnrc_ipv6_nib: add address from netif to address validation timer
+    + netdev_ieee802154: add txpower and page
+    * gnrc_sixlowpan_frag: allow send of multiple datagrams simultaneously
```
Comment (miri64) [inline release-notes.txt]: Sorry I wasn't clear on that, but I also meant this line with https://github.com/RIOT-OS/RIOT/pull/11901#discussion_r306795543

Comment (MrKevinWeiss) [reply release-notes.txt]: Thanks for clarifying!

**Comment by miri64** [comment]
Please squash.

**Comment by MrKevinWeiss** [comment]
Many thanks for your review!

**Pull request merged**